segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/fs/FSStorage.java

```diff
@@ -0,0 +1,382 @@
+    }
+
+    @Override
+    public CompletableFuture<Void> write(SegmentHandle handle, long offset, InputStream data, int length, Duration
+            timeout) {
+        return CompletableFuture.supplyAsync(() -> syncWrite(handle, offset, data, length, timeout), executor);
+    }
+
+    @Override
+    public CompletableFuture<Void> seal(SegmentHandle handle, Duration timeout) {
+        return CompletableFuture.supplyAsync(() -> syncSeal(handle, timeout), executor);
+    }
+
+    @Override
+    public CompletableFuture<Void> concat(SegmentHandle targetHandle, long offset, String sourceSegment, Duration
+            timeout) {
+            return CompletableFuture.supplyAsync(() -> syncConcat(targetHandle, offset, sourceSegment, timeout),
+                    executor);
+    }
+
+    @Override
+    public CompletableFuture<Void> delete(SegmentHandle handle, Duration timeout) {
+            return CompletableFuture.supplyAsync(() -> syncDelete(handle, timeout), executor);
+    }
+
+    //endregion
+
+    //region AutoClosable
+
+    @Override
+    public void close() {
+
+    }
+
+    //endregion
+
+    //region private sync implementation
+
+    private SegmentHandle syncOpenRead(String streamSegmentName) {
+        Path path = Paths.get(config.getNfsRoot(), streamSegmentName);
+
+        if (!Files.exists(path)) {
+            throw new CompletionException(new StreamSegmentNotExistsException(streamSegmentName));
+        }
+
+        return FSSegmentHandle.getReadHandle(streamSegmentName);
+    }
+
+
+    private int syncRead(SegmentHandle handle, long offset, byte[] buffer, int bufferOffset, int length, Duration
+            timeout) {
+
+        Path path = Paths.get(config.getNfsRoot(), handle.getSegmentName());
+
+        if (!Files.exists(path)) {
+            throw new CompletionException(new StreamSegmentNotExistsException(handle.getSegmentName(), null));
+        }
+
+        try {
+            if (Files.size(path) < offset) {
+                log.warn("Read called on segment {} at offset {}. The offset is beyond the current size of the file.",
+                        handle.getSegmentName(), offset);
+                throw new CompletionException(new IllegalArgumentException());
+            }
+        } catch (IOException ioe) {
+            throw new CompletionException(ioe);
+        }
+
+        try (FileChannel channel = FileChannel.open(path, StandardOpenOption.READ)) {
+            int bytesRead = channel.read(ByteBuffer.wrap(buffer, bufferOffset, length), offset);
+            return bytesRead;
+        } catch (IOException | IndexOutOfBoundsException e) {
+            if (e instanceof IndexOutOfBoundsException) {
+                throw new CompletionException(new ArrayIndexOutOfBoundsException(e.getMessage()));
+            } else {
+                throw new CompletionException(e);
+            }
+        }
+    }
+
+    private SegmentProperties syncGetStreamSegmentInfo(String streamSegmentName, Duration timeout) {
```

Comment:
Here and on all of the other sync methods there is a parameter: timeout that is never used. This is misleading and should be removed.